sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/EncodingPerFile.java

```diff
@@ -39,16 +38,16 @@ public EncodingPerFile(AbstractGlobalProtobufFileProcessor globalReportProcessor
   }
 
   boolean encodingMatch(InputFile inputFile) {
-    URI inputFileUri = inputFile.uri();
+    String uri = inputFile.uri().toString();
 
-    if (!globalReportProcessor.getRoslynEncodingPerUri().containsKey(inputFileUri)) {
+    if (!globalReportProcessor.getRoslynEncodingPerUri().containsKey(uri)) {
       // When there is no entry for a file, it means it was not processed by Roslyn. So we consider encoding to be ok.
       return true;
     }
 
-    Charset roslynEncoding = globalReportProcessor.getRoslynEncodingPerUri().get(inputFileUri);
+    Charset roslynEncoding = globalReportProcessor.getRoslynEncodingPerUri().get(uri); // Case-Insensitive check
```

Comment:
SC says there's a code smell here, and I tend to agree - the line is quite long, it's better to put the comment above